openedx/frontend-app-authoring
# chore: iframe rendering optimization

🚨 **Dependencies:**
- 🔗 [edx-plarform PR](https://github.com/openedx/edx-platform/pull/35959)

## Description

This PR provides iframe reload optimizations for various xblock related actions. Added some improvements related to scrolling to the current xblock. Fixed behavior of the xblock action dropdown list.

## Testing instructions

1. Open New Course Unit page.
2. When duplicating a block, a scroll to a new copy of the block occurs.
3. Delete, duplicate, drag xblock - a full iframe reboot should not occur.
4. Open a new xblock editor, save the changes - after returning to the course unit page, the page should scroll to the xblock for which the changes were made.

https://github.com/user-attachments/assets/e32a9fa7-09f1-4d5e-b616-82368030b3f1

## Other information

Improvements to the editing modals and tagging will be added in future PRs.

**Settings**
```yaml
EDX_PLATFORM_REPOSITORY: https://github.com/raccoongang/edx-platform.git
EDX_PLATFORM_VERSION: Peter_Kulko/rendering-optimization

TUTOR_GROVE_NEW_MFES:
  authoring:
    port: 18000
    repository: https://github.com/raccoongang/frontend-app-course-authoring.git
    version: Peter_Kulko/rendering-optimization

TUTOR_GROVE_WAFFLE_FLAGS:
  - name: contentstore.new_studio_mfe.use_new_unit_page
    everyone: true

TUTOR_GROVE_MFE_LMS_COMMON_SETTINGS:
  MFE_CONFIG:
    ENABLE_UNIT_PAGE: true
```
